azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/deploy/FTPFunctionDeployHandler.java

```diff
@@ -4,13 +4,37 @@
  */
 package com.microsoft.azure.toolkit.lib.appservice.service.impl.deploy;
 
+import com.azure.resourcemanager.appservice.models.PublishingProfile;
 import com.azure.resourcemanager.appservice.models.WebAppBase;
+import com.microsoft.azure.management.appservice.FunctionApp;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureExecutionException;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.logging.Log;
+import com.microsoft.azure.toolkit.lib.legacy.appservice.handlers.artifact.FTPUploader;
 
 import java.io.File;
 
 public class FTPFunctionDeployHandler implements IFunctionDeployHandler {
+    private static final String DEFAULT_WEBAPP_ROOT = "/site/wwwroot";
+    private static final int DEFAULT_MAX_RETRY_TIMES = 3;
+
     @Override
-    public void deploy(File file, WebAppBase functionApp) {
+    public void deploy(final File file, final WebAppBase webAppBase) {
+        final FTPUploader uploader = new FTPUploader();
+        final PublishingProfile profile = webAppBase.getPublishingProfile();
+        final String serverUrl = profile.ftpUrl().split("/", 2)[0];
+
+        try {
+            uploader.uploadDirectoryWithRetries(serverUrl, profile.ftpUsername(), profile.ftpPassword(),
+                    file.getAbsolutePath(), DEFAULT_WEBAPP_ROOT, DEFAULT_MAX_RETRY_TIMES);
+        } catch (AzureExecutionException e) {
```

Comment:
why not throw AzureToolkitRuntimeException directly from the upload method.

Reply:
Legacy codes shared with intellij/gradle, will refactor and deprecate `AzureExecutionException` later